modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/LeaseUpdater.java

```diff
@@ -210,39 +237,88 @@ public void run() {
-            var newTs = new HybridTimestamp(clock.now().getPhysical() + LEASE_PERIOD, 0);
+
+            HybridTimestamp startTs = clock.now();
+
+            var expirationTs = new HybridTimestamp(startTs.getPhysical() + longLeasePeriod, 0);
 
             byte[] leaseRaw = ByteUtils.toBytes(lease);
 
-            Lease renewedLease = new Lease(candidate, newTs);
+            Lease renewedLease = new Lease(candidate, startTs, expirationTs);
 
             msManager.invoke(
                     or(notExists(leaseKey), value(leaseKey).eq(leaseRaw)),
                     put(leaseKey, ByteUtils.toBytes(renewedLease)),
                     noop()
+            ).thenAccept(isCreated -> {
+                if (isCreated) {
+                    boolean force = candidate.equals(lease.getLeaseholder());
+
+                    leaseConciliator.conciliate(grpId, renewedLease, force);
+                }
+            });
+        }
+
+        /**
+         * Writes a prolong lease in Meta storage.
+         *
+         * @param grpId Replication group id.
+         * @param lease Lease to prolong.
+         */
+        private void prolongLeaseInMetaStorage(ReplicationGroupId grpId, Lease lease) {
+            var leaseKey = ByteArray.fromString(PLACEMENTDRIVER_PREFIX + grpId);
+            var newTs = new HybridTimestamp(clock.now().getPhysical() + LEASE_PERIOD, 0);
+
+            byte[] leaseRaw = ByteUtils.toBytes(lease);
+
+            Lease renewedLease = lease.prolongLease(newTs);
+
+            msManager.invoke(
+                    value(leaseKey).eq(leaseRaw),
+                    put(leaseKey, ByteUtils.toBytes(renewedLease)),
+                    noop()
+            );
+        }
+
+        /**
+         * Writes an accepted lease in Meta storage.
+         *
+         * @param grpId Replication group id.
+         * @param lease Lease to accept.
+         */
+        private void acceptLeaseInMetaStorage(ReplicationGroupId grpId, Lease lease) {
+            var leaseKey = ByteArray.fromString(PLACEMENTDRIVER_PREFIX + grpId);
+            var newTs = new HybridTimestamp(clock.now().getPhysical() + LEASE_PERIOD, 0);
+
+            byte[] leaseRaw = ByteUtils.toBytes(lease);
+
+            Lease renewedLease = lease.acceptLease(newTs);
+
+            msManager.invoke(
+                    value(leaseKey).eq(leaseRaw),
+                    put(leaseKey, ByteUtils.toBytes(renewedLease)),
+                    noop()
             );
         }
 
         /**
-         * Checks that a leaseholder candidate can take a lease on the replication group.
+         * Checks that the lease is outdated.
          *
          * @param lease Lease.
-         * @param candidate The node is a leaseholder candidate.
          * @return True when the candidate can be a leaseholder, otherwise false.
          */
-        private boolean isReplicationGroupUpdateLeaseholder(Lease lease, ClusterNode candidate) {
+        private boolean isLeaseIsOutdated(Lease lease) {
```

Comment:
Actually, this returns true not only if the lease is outdated, but when there is no lease at all (due to EMPTY_LEASE). Please rename this to `isLeaseIsOutdatedOrNonExistent` and explain in javadoc why the logic for both cases should be the same: we can't prolong the expired lease because we already have an interval of time when the lease was not active, so we must start ne negotiation round from the beginning; the same we do for the groups that don't have leaseholders at all.